arch/eventloop_posix_udp.c

```diff
@@ -36,6 +44,524 @@ typedef struct {
+                UA_Boolean enableLoopback = *(UA_Boolean*)prop->value.data;
+                UA_StatusCode res = setLoopBackData(socket, enableLoopback, ai_family, logger);
+                if (res != UA_STATUSCODE_GOOD) return res;
+            }
+        } else if(UA_String_equal(&prop->key.name, &reuseParam)) {
+            if(UA_Variant_hasScalarType(&prop->value, &UA_TYPES[UA_TYPES_BOOLEAN])) {
+                UA_Boolean enableReuse = *(UA_Boolean *)prop->value.data;
+                setReuseAddress(socket, enableReuse, logger);
+            }
+#ifdef __linux__
+        } else if(UA_String_equal(&prop->key.name, &socketPriorityParam)){
+            if(UA_Variant_hasScalarType(&prop->value, &UA_TYPES[UA_TYPES_UINT32])){
+                UA_UInt32 *socketPriority = (UA_UInt32 *) UA_malloc(sizeof(UA_UInt32));
+                if(!socketPriority) {
+                    UA_LOG_WARNING(logger, UA_LOGCATEGORY_SERVER,
+                                   "PubSub Connection creation. Could not set socket priority due to out of memory.");
+                    continue;
+                }
+                UA_UInt32_copy((UA_UInt32 *) prop->value.data, socketPriority);
+                setSocketPriority(socket, socketPriority, logger);
+                UA_free(socketPriority);
+            }
+#endif
+        } else if (UA_String_equal(&prop->key.name, &hostnameParam) ||
+                   UA_String_equal(&prop->key.name, &portParam) ||
+                   UA_String_equal(&prop->key.name, &listenHostnamesParam) ||
+                   UA_String_equal(&prop->key.name, &listenPortParam)) {
+            /* ignore, required args are handled elsewhere explicitly */
+        } else {
+            UA_LOG_WARNING(logger, UA_LOGCATEGORY_SERVER,
+                           "PubSub Connection creation. Unknown connection parameter.");
+        }
+    }
+    /* Set the socket options
+    * also there might be external requirements for socket options
+    * how to handle and set those
+    * either via: passed in parameters
+    * or via: specific callback that gives access to the allocated socket
+    * */
+    UA_StatusCode res = UA_STATUSCODE_GOOD;
+    res |= UDP_setNonBlocking(socket);
+    res |= UDP_setNoSigPipe(socket);
+    if(res != UA_STATUSCODE_GOOD) {
+        UA_LOG_SOCKET_ERRNO_WRAP(
+            UA_LOG_WARNING(logger, UA_LOGCATEGORY_NETWORK,
+                           "UDP\t| Could not set socket options: %s", errno_str));
+        UA_close(socket);
+        return res;
+    }
+
+    return UA_STATUSCODE_GOOD;
+}
+static UA_StatusCode
+configureMulticastInterfaceIPv4(UA_FD socket,
+                                socklen_t addrlen, struct sockaddr_in *addr,
+                                size_t paramsSize, const UA_KeyValuePair *params, const UA_Logger *logger){
+    struct in_addr address = addr->sin_addr;
+    UA_Byte *addressVal = (UA_Byte *) &addr->sin_addr;
+
+    if(isIPv4MulticastAddress(addressVal)) {
```

Comment:
The name of the function is misleading.
You don’t know whether it is a multicast address before calling the method.
Check this outside and only call this method of it is a multicast address.
The same applies to the below functions.